File: apple/Sources/ReactNativeWatchOS/Modifiers/Monospaced.swift
import SwiftUI

/// SwiftUI `.monospaced(_:)`. Renders text within the view with a fixed-
/// width (monospaced) font. Defaults to active when `value` is omitted.
enum RNWMonospacedModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("monospaced") { view, params, _ in
            AnyView(view.monospaced(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/monospaced.ts
import { createModifier } from './createModifier';

export interface MonospacedParams {
  /// Whether the monospaced font is active. Defaults to `true`.
  value?: boolean;
}

/// SwiftUI `.monospaced(_:)`. Renders text within the view with a fixed-
/// width (monospaced) font. Pass `false` to explicitly turn it off.
export function monospaced(params?: MonospacedParams | boolean) {
  if (typeof params === 'boolean') {
    return createModifier('monospaced', { value: params });
  }
  return createModifier('monospaced', params);
}
